ta/Makefile

```diff
@@ -5,6 +5,8 @@ ifeq ($O,)
 $(error output path should be specified when calling this makefile)
 endif
 
+LDFLAGS=
```

Comment:
Sumit, your suggestion is working, i.e, the following change on top of your patch (and I think this is more clean than the suggestion from @jforissier above).
```
diff --git a/Makefile b/Makefile
index e199ff9..f4d8622 100644
--- a/Makefile
+++ b/Makefile
@@ -25,6 +25,7 @@ endif
 .PHONY: xtest
 xtest:
        $(q)$(MAKE) -C host/xtest CROSS_COMPILE="$(CROSS_COMPILE_HOST)" \
+                            --no-builtin-variables \
                             q=$(q) \
                             O=$(out-dir)/xtest \
                             $@
```

So, then that would be OK I think. But then we still have the remaining question about the LDFLAGS. How did you call when you tried Jeromes suggestion?
  `make LDFLAGS=`
or
  `LDFLAGS= make`
?